File: main/maximum-subarray/maximum-subarray.cpp
class Solution {
public:
    static int maxSubArray(const vector<int>& nums) noexcept;
};

int Solution::maxSubArray(const vector<int>& nums) noexcept
{
    auto acc = numeric_limits<int>::min();
    auto cur = 0;

    for (const auto elem : nums) {
        cur = max(cur, 0) + elem;
        acc = max(acc, cur);
    }

    return acc;
}

File: main/maximum-subarray/maximum-subarray.py
class Solution:
    def maxSubArray(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        best_total = -math.inf

        total = 0
        for x in nums:
            total += x
            if best_total < total:
                best_total = total
            if total < 0:
                total = 0

        return best_total
